### ISSUE TITLE
Error handling

### ISSUE THREAD

**Issue by scizzorz** [scizzorz/rain#9]
My original idea is to just have a bunch of falsy named singletons for different errors like a missing table key, invalid arithmetic, etc but I'll need to think about this a lot.

**Comment by scizzorz** [comment]
Silently failing can cause a lot of tricky bugs, like accidentally comparing a number and a table instead of comparing a number and a table's attribute.

**Comment by scizzorz** [comment]
The framework is implemented as of 7b3d4362a9ba9d01df1e90087119f8bcf3e3343d.

You can raise an exception with the panic builtin:

```
panic("Error!")
```

You can catch an unwinding panic with a catch call:
```
let x = might_throw_error?(...) # will either return the correct value or the exception raised anywhere down the line
```

And you can use a blanket catch block that binds an unwinding panic to a variable:

```
catch x # will bind null to x if no panic happens, or the caught exception
  might_throw_error(...)
```
